File: Java/495.java
__________________________________________________________________________________________________
sample 1 ms submission
class Solution {
    public int findPoisonedDuration(int[] timeSeries, int duration) {
        int res = 0;
        if (timeSeries.length == 0) return res;
        int next = 0;
        for (int i : timeSeries) {
            if (i < next) {
                res -= next - i;
                
            }
next = i + duration;
                res += duration;
        }
        return res;
    }
}
__________________________________________________________________________________________________
sample 38920 kb submission
class Solution {
    public int findPoisonedDuration(int[] timeSeries, int duration) {
        int res = 0;
        if(timeSeries == null || timeSeries.length == 0)
            return res;
        for(int i = 1; i < timeSeries.length; i++) {
            if(timeSeries[i]-duration > timeSeries[i-1])
                res += duration;
            else
                res += timeSeries[i]-timeSeries[i-1];
        }
        return res+duration;
    }
}
__________________________________________________________________________________________________

File: Python3/495.py
__________________________________________________________________________________________________
sample 264 ms submission
class Solution:
    def findPoisonedDuration(self, timeSeries: List[int], duration: int) -> int:
        if not timeSeries:
            return 0
        
        total_duration = 0
        current_end = timeSeries[0]
        for attack in timeSeries:
            attack_end = attack + duration
            if attack > current_end:
                total_duration += duration
            else:
                total_duration += attack_end - current_end
            current_end = attack_end
                
        return total_duration
__________________________________________________________________________________________________
sample 268 ms submission
class Solution:
    def findPoisonedDuration(self, timeSeries: List[int], duration: int) -> int:
        res = 0
        if not timeSeries: return 0
        prev = timeSeries[0]
        for value in timeSeries[1:]:
            if value > prev+duration:
                res += duration
            else:
                res += value-prev
            prev = value
        return res+duration
__________________________________________________________________________________________________
